rpcs3/rpcs3qt/gamepads_settings_dialog.cpp

```diff
@@ -154,17 +126,38 @@ gamepads_settings_dialog::gamepads_settings_dialog(QWidget* parent)
 	setLayout(dialog_layout);
 	layout()->setSizeConstraint(QLayout::SetFixedSize);
 
-	configure_combos();
+	auto configure_combos = [=]
+	{
+		//Set the values from config
+		for (int i = 0; i < MAX_PLAYERS; i++)
+		{
+			// No extra loops are necessary because setCurrentText does it for us
+			co_inputtype[i]->setCurrentText(qstr(input_cfg.player_input[i].to_string()));
+			// Device will be empty on some rare occasions, so fill them by force
+			ChangeInputType(i);
+		}
+	};
 
 	for (int i = 0; i < MAX_PLAYERS; i++)
 	{
 		connect(co_inputtype[i], &QComboBox::currentTextChanged, [=] { ChangeInputType(i); });
-		connect(co_deviceID[i], &QComboBox::currentTextChanged, [=] { ChangeDevice(i); });
+		connect(co_deviceID[i], &QComboBox::currentTextChanged, [=](const QString& dev)
```

Comment:
IMHO, it's better to be epxlicit, when capturing, especially when the capture list is small. It's easier to spot bug this way (when something is captured by value, when it should be by reference or vice-versa). Most (if not all) of this `[=]` connect captures can be changed to just `[i]`, `[this]` or `[this, i]`.

Reply:
I'm doing this specifically because I had bugs before that, in the end, were caused by such things.
So I'd prefer to leave it as is right now.
Feel free to change these things in a follow up PR if you don't mind. I am tired of testing the code for every minute change.

Reply:
Because you probably connected to lambda that captured locals by reference. Or you forgot to add context parameter (to the connect), so the lambda-slot was called even after the associated object was destroyed (though in that case it'd still crash in most cases). Doesn't matter though.

Reply:
yeah I had all of these at least once. And I also had a very weird follow up error that took me forever to backtrack until I ended up at some lambda with wrong capture hehe. Therefore I go the safe route first before I end up chasing ghosts.